### ISSUE TITLE
Building on OS X now copies, removes cside-choicescript directory

### ISSUE THREAD

**Issue by KLNeidecker** [ChoicescriptIDE/main#16]
(Thanks for looking into the last issue)

Pulled the 'latest' branch to see if the changes allowed a better compile. When I begin the grunt process, it downloads cside-choicescript, and everything looks good, until it stops with:

```
fs.js:653
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open 'build/node_modules/cside-choicescript/web/mygame/index.html'
    at Object.fs.openSync (fs.js:653:18)
    at Object.fs.readFileSync (fs.js:554:33)
    at slurpFile (/Users/kristopherneidecker/ChoiceScript IDE/create-compile-html.js:12:17)
    at compile (/Users/kristopherneidecker/ChoiceScript IDE/create-compile-html.js:25:19)
    at Object.<anonymous> (/Users/kristopherneidecker/ChoiceScript IDE/create-compile-html.js:89:12)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
Warning: -> error 1 /Users/kristopherneidecker/ChoiceScript IDE/create-compile-html.js (155ms) Use --force to continue.
```

And when looking into the build/node_modules the cside-choicescript directory that was there is gone. When manually copying said directory back in and running the process again, it deletes it, and then chokes since it can't find the files it deleted.

(edited for spelling. Yay spelling!)

**Comment by CareyJWilliams** [comment]
Delete **everything**, re-clone and ensure you're using a relatively modern NodeJS + NPM (6+ I'm using, I think). You might have noticed I've added an automated build process to the repo, after that last slip up. The build is currently passing, so I'm pretty confident this isn't a direct build issue.

I had a problem with missing commits when I moved from master -> latest as the default branch, and had to delete and start fresh (might be related). Let me know!

**Comment by KLNeidecker** [comment]
Node version: v8.6.0
NPM: 5.4.2

Deleted. Cloned new. Made a directory "main". Builds fine now, but the "graphical" issue is still there. Same as before, where the UI is munged and unusable.

Not sure what it could be on my end making it create an unusable UI.

OS X 10.13, Homebrew 1.3.5, Homebrew/homebrew-core (git revision c7fb0c; last commit 2017-10-09)

**Comment by CareyJWilliams** [comment]
You should be able to right-click -> inspect element and pull up a console, just like in a web browser. There will likely be an error there.

**Comment by KLNeidecker** [comment]
Ahh, here:

```
module.js:496 Uncaught Error: Cannot find module 'cside-updater'
    at Function.Module._resolveFilename (module.js:494:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:522:17)
    at require (internal/module.js:20:19)
    at self.require (<anonymous>:11:26)
win_state.min.js:2 Uncaught TypeError: Cannot read property 'isDevToolsOpen' of undefined
    at initWindowState (win_state.min.js:2)
    at win_state.min.js:31
node_CSIDE.min.js:65 Uncaught TypeError: Cannot read property 'on' of undefined
    at node_CSIDE.min.js:65
```

**Comment by CareyJWilliams** [comment]
OK. That's odd. cside-updater should come with the source under source/node_modules. Could you check if it's included in your build folder, under node_modules? I'm thinking this might be a node v6 vs node v8 thing.

**Comment by KLNeidecker** [comment]
It is. 2 files: index.js, package.json. The other dir there is cside-storage.

p.s. Thanks for being so responsive. I have a big project I want to work on and this IDE kicks butt. Been pining to use the fixed sound functions and the like, hence my attempts at compiling (and the grunt situation is new to me, as my coding happens in IDEs for the most part, and I've avoided much of the make/grunt type systems until recently)

**Comment by CareyJWilliams** [comment]
No worries, I'm just as eager to get to the bottom of this! Saying that, you can get the fixed sound functions (and any frequent updates) via the new 'Development' update channel. No need to compile from source anymore! :)

**Comment by KLNeidecker** [comment]
Excellent. Going to ask a dumb question: how do I access said channel ;)

So, I hard reset the dir, and started from scratch. Here is the log:
```
removed 125 packages in 4.968s
Running "clean:files" (clean) task
>> 0 paths cleaned.

Running "clean:folders" (clean) task
>> 8 paths cleaned.

Running "auto_install:codemirror" (auto_install) task
running npm install  on node_modules/cside-codemirror...
> phantomjs-prebuilt@2.1.15 install /Users/<redacted ;)>/main/node_modules/cside-codemirror/node_modules/phantomjs-prebuilt
> node install.js

PhantomJS not found on PATH
Download already available at /var/folders/ys/k_ck6r216lq8nfb4m2n6ybd40000gn/T/phantomjs/phantomjs-2.1.1-macosx.zip
Verified checksum of previously downloaded file
Extracting zip contents
Removing /Users/<redacted ;)>/main/node_modules/cside-codemirror/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /var/folders/ys/k_ck6r216lq8nfb4m2n6ybd40000gn/T/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1507665020107/phantomjs-2.1.1-macosx -> /Users/<redacted ;)>/main/node_modules/cside-codemirror/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /Users/<redacted ;)>/main/node_modules/cside-codemirror/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs

> cside-codemirror@5.23.0 prepublish /Users/<redacted ;)>/main/node_modules/cside-codemirror
> npm run-script build


> cside-codemirror@5.23.0 build /Users/<redacted ;)>/main/node_modules/cside-codemirror
> rollup -c

added 125 packages in 7.128s
[32m✔[0m

Running "copy:main" (copy) task
Created 20 directories, copied 144 files

Running "copy:choicescript" (copy) task
Created 7 directories, copied 59 files

Running "concat:dist" (concat) task

Running "uglify:my_target" (uglify) task
>> 4 files created 1.33 MB → 1.39 MB

Running "cssmin:target" (cssmin) task
>> 1 file created. 211.45 kB → 185.88 kB

Running "auto_install:build" (auto_install) task
running npm install trash username on build...+ username@1.0.1
+ trash@2.0.0
added 126 packages and removed 2 packages in 2.455s
[32m✔[0m

Running "string-replace:trashPatch" (string-replace) task

1 files created

Running "execute:target" (execute) task
-> executing /Users/<redacted ;)>/main/create-compile-html.js

Extracting js data from:
../version.js
../persist.js
../alertify.min.js
../util.js
../ui.js
../scene.js
../navigator.js
mygame.js


Extracting css data from:
../style.css
../alertify.css
Assembling new html file...
DONE!
-> completed /Users/<redacted ;)>/main/create-compile-html.js (125ms)

>> 1 file and 0 calls executed (131ms)

Running "compress:main" (compress) task
>> Compressed 1151 files.

Running "nwjs:mac" (nwjs) task
Using v0.21.4
Create cache folder in /Users/<redacted ;)>/main/nwjsCache/0.21.4
Using cache for: osx64
Create release folder in /Users/<redacted ;)>/main/nwjsBuild/CSIDE/osx64
>> NW.js app created.

Done.

```

And the following wasn't piped to the text file and showed up in the term:
```
npm WARN grunt-execute@0.2.2 requires a peer of grunt@~0.4.1 but none is installed. You must install peer dependencies yourself.
npm WARN CSIDE@1.1.2 No repository field.
npm WARN CSIDE@1.1.2 No license field.


(node:18732) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
npm WARN deprecated osx-trash@1.0.1: Deprecated in favor of the `trash` module
npm WARN deprecated win-trash@1.0.1: Deprecated in favor of the `trash` module
npm WARN CSIDE@1.1.2 No description
npm WARN CSIDE@1.1.2 No repository field.
npm WARN CSIDE@1.1.2 No license field.
```

**Comment by CareyJWilliams** [comment]
Hmm, nothing obviously suspicious there. Just open up the production version of CSIDE, go to the settings tab, scroll to the bottom and it's an option under "Update Channels" :)

**Comment by KLNeidecker** [comment]
Ahh great, I see it. Looking good, and it runs. I'm still here to test changes, so just let me know if more steps are needed.

**Comment by CareyJWilliams** [comment]
OK, @KLNeidecker, this should be fixed now. Recent versions of NPM seem to remove any modules already in the node_modules folder when npm install is called. I'm not sure if this is a bug or by design, but previous versions did not. At any rate, the latest commit breaks the build down a bit and shuffles the order (copying the modules after the problematic install call). If you get a moment to try, I'd really appreciate a +1 on whether or not this is finally sorted.

**Comment by KLNeidecker** [comment]
Working great. Compiled and ran fine!

**Comment by CareyJWilliams** [comment]
Cheers!

Closed as resolved by 5bea00220a96b8be883d42b32fd1a7dfcba9a69b
